lms/djangoapps/discussion_api/serializers.py

```diff
@@ -195,19 +205,24 @@ class ThreadSerializer(_ContentSerializer):
     def __init__(self, *args, **kwargs):
         remove_fields = kwargs.pop('remove_fields', None)
         super(ThreadSerializer, self).__init__(*args, **kwargs)
-        # type is an invalid class attribute name, so we must declare a
-        # different name above and modify it here
-        self.fields["type"] = self.fields.pop("type_")
         # Compensate for the fact that some threads in the comments service do
         # not have the pinned field set
-        if self.object and self.object.get("pinned") is None:
-            self.object["pinned"] = False
-
+        if self.instance and self.instance.get("pinned") is None:
+            self.instance["pinned"] = False
+        if self.instance and self.instance.get("resp_total") is None:
+            self.instance["resp_total"] = None
         if remove_fields:
             # for multiple fields in a list
             for field_name in remove_fields:
                 self.fields.pop(field_name)
 
+    def get_pinned(self, obj):
+        """
+        Compensate for the fact that some threads in the comments service do
+        not have the pinned field set.
+        """
+        return bool(obj["pinned"])
+
```

Comment:
@cpennington above are the changes we discussed from the previous PR:

- Self.object is now self.instance: I have changed this and the code guarantees that a default will be set for both pinned and response_count
    - this ensures that when get_pinned is called that the obj dictionary will not be null attempting to get the value at "pinned" https://github.com/edx/edx-platform/blob/bbeggs/merge-patch-2015-09-22/lms/djangoapps/discussion_api/serializers.py#L227
- There is a test that exercises the pinned k/v pair is missing here: https://github.com/edx/edx-platform/blob/bbeggs/merge-patch-2015-09-22/lms/djangoapps/discussion_api/tests/test_serializers.py#L231-L240
- I have added a test that checks to make sure that response_count shows up in the serialized object with a value of None even if it is not in the JSON. def test_response_count_missing(self):
- I have removed the line `self.fields["type"] = self.fields.pop("type_")` as "type" is no longer a protected  keyword